eparticlectr/eparticlectr.cpp

```diff
@@ -433,6 +447,154 @@ void eparticlectr::finalize( uint64_t proposal_id ) {
         proptable.erase( prop_it );
 }
 
+[[eosio::action]]
+void eparticlectr::finalizeextr( uint64_t proposal_id ) {
+    // Verify proposal exists
+    propstblex proptable( _self, _self.value );
+    auto prop_it = proptable.find( proposal_id );
+    eosio::check( prop_it != proptable.end(), "Proposal not found" );
+
+    // Verify voting period is complete
+    eosio::check( eosio::current_time_point().sec_since_epoch() > prop_it->endtime, "Voting period is not over yet");
+
+    // Retrieve votes from DB
+    votestblex votetbl( _self, proposal_id );
+    auto vote_it = votetbl.begin();
+    eosio::check( vote_it != votetbl.end(), "No votes found for proposal");
+
+    // Tally votes
+    uint64_t yes_votes = 0;
+    uint64_t no_votes = 0;
+    while(vote_it->proposal_id == proposal_id && vote_it != votetbl.end()) {
```

Comment:
since you are getting the table for the scope proposal_id ( votestblex votetbl( _self, proposal_id ) ) I guess all the vote_it should belong to this proposal, so this check might not be needed.

this might not scale if in the future voting gets more than 100 - 150 users voting as the while might timeout. I guess we can deploy this, and if we ever reach that point we could improve this. Probably it should be needed a table that stores total votes so you dont need to iterate over this.

Reply:
Will leave for now, as adding a table for votes would increase ram.